pcsx2/gui/GlobalCommands.cpp

```diff
@@ -150,15 +150,20 @@ namespace Implementations
 			OSDlog( Color_StrongRed, true, "(FrameLimiter) SlowMotion ENABLED." );
 			g_Conf->EmuOptions.GS.FrameLimitEnable = true;
 		}
+
+		GSsetVsync(EmuConfig.GS.GetVsync());
+
 		pauser.AllowResume();
 	}
 
 	void Framelimiter_MasterToggle()
 	{
 		ScopedCoreThreadPause pauser;
 		g_Conf->EmuOptions.GS.FrameLimitEnable = !g_Conf->EmuOptions.GS.FrameLimitEnable;
-		GSsetVsync( g_Conf->EmuOptions.GS.FrameLimitEnable && g_Conf->EmuOptions.GS.VsyncEnable );
 		OSDlog( Color_StrongRed, true, "(FrameLimiter) %s.", g_Conf->EmuOptions.GS.FrameLimitEnable ? "ENABLED" : "DISABLED" );
+
+		GSsetVsync(EmuConfig.GS.GetVsync());
```

Comment:
Hum, interesting, behavior wasn't the same between GSopen and GSopen2.

Still something annoye me, hitting turbo/slowmo will enable the framelimiter. But it will never be disabled if you left the turbo/slowmo mode. I understand that turbo/slowmo don't make sense without a frame limiter. But I'm not sure toggling framelimiter is a good idea.

So is there any case that we want to disable Vsync. We have the turbo mode (but it is likely useless for high fps display). Do we want to disable Vsync when the user hit F4 to disable the frame limiter ? My feeling is no. And then there is the magical combination.
* no frame limiter + vsync
* hit turbo -> enable frame limiter / disable vsync
* hit f4 -> no frame limiter / vsync still off / turbo still on